plotly/plotly/plotly.py

```diff
@@ -1308,7 +1308,14 @@ def add_share_key_to_url(plot_url):
     body = {'share_key_enabled': True, 'world_readable': False}
     response = v2.files.update(fid, body)
 
-    return plot_url + '?share_key=' + response.json()['share_key']
+    if not v2.files.retrieve(fid).json()['share_key_enabled']:
+        attempt += 1
+        if attempt == 10:
+            raise Exception
```

Comment:
We're trying to wrap everything in plotly exceptions so that users can catch them. So just: `raise exceptions.PlotlyException` (or something)